src/main/java/org/vaadin/viritin/ListContainer.java

```diff
@@ -482,15 +485,16 @@ private PropertyComparator(Object[] propertyId, boolean[] ascending) {
         public int compare(T o1, T o2) {
             for (int i = 0; i < propertyId.length; i++) {
                 String currentProperty = propertyId[i].toString();
-                Comparator<T> currentComparator
-                        = new BeanComparator<T>(currentProperty,
-                                getUnderlyingComparator(currentProperty));
+                Comparator underlyingComparator = getUnderlyingComparator(currentProperty);
+                Comparator currentComparator = underlyingComparator != null ? underlyingComparator : ComparableComparator.getInstance();
 
                 if (!ascending[i]) {
-                    currentComparator = new ReverseComparator(currentComparator);
+                    currentComparator = currentComparator.reversed();
                 }
 
-                int compare = currentComparator.compare(o1, o2);
+                Object o1Prop = getContainerProperty(o1, currentProperty).getValue();
```

Comment:
Not sure, but should check if this is in par with the old version performance vice. Also I'd more rely on BeanUtils API than on the Vaadin's Property API. Have you tried if BeanComparator supports nested property? I think the actual thing we are trying to achieve with this patch doesn't require this change.